### PR TITLE
feat: Adds Svelte 5 support

### LINKED ISSUE #1736: Support Svelte v5 beta
### Package

- [ ] lucide
- [ ] lucide-angular
- [ ] lucide-flutter
- [ ] lucide-preact
- [ ] lucide-react
- [ ] lucide-react-native
- [ ] lucide-solid
- [X] lucide-svelte
- [ ] lucide-vue
- [ ] lucide-vue-next
- [ ] Figma plugin
- [ ] all JS packages
- [ ] site

### Description

Cannot install lucide-svelte in a Svelte 5 project

image

### Use cases

Using Svelte 5, which should be API compatible for something like Lucide

### Checklist

- [X] I have searched the existing issues to make sure this bug has not already been reported.

### PR DESCRIPTION
**Pull request by AdrianGonz97**
closes #1736 

## What is the purpose of this pull request?

- [ ] New Icon
- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other:

### Description
This PR adds support for the Svelte 5 beta. I changed `>=3 <5` to `^3 || ^4` as they are functionally equivalent and appended the beta version to the end: `|| ^5.0.0-next.0`.

It's important to note that there is an open issue in Svelte 5 with svgs failing to render on the client: https://github.com/sveltejs/svelte/issues/9645. 

As such, you may want to consider holding back merging this PR until that issue is resolved (perhaps to prevent folks from mistakingly thinking that this bug is originating from the library).

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by ericfennis** [comment]
Hmm, I've tested this. But icons seem not rendered. Weirdly the HTML content is present in the DOM. But nothing is shown, and no CSS that maybe hides icon is applied. Same in Safari. Is there a known Svelte 5 issue with SVG rendering?
image

**Comment by ericfennis** [comment]
Hmm not sure if should merge this if icons SVGs are not visible 🤔

**Comment by AdrianGonz97** [comment]
@ericfennis Agreed. Best to hold off until the issue is fixed!

**Comment by ericfennis** [comment]
@AdrianGonz97 Looks like they fixed it! 🎉 
image

Fixed in https://github.com/sveltejs/svelte/pull/10006

File: packages/lucide-svelte/package.json
Comment (ericfennis) [inline packages/lucide-svelte/package.json:74]: Maybe we should specify which version it was fixed?

Comment (AdrianGonz97) [reply packages/lucide-svelte/package.json]: Good catch!

**Review by ericfennis — APPROVED** [review]

**Pull request merged**
